test/avocado/run-tests

```diff
@@ -134,21 +134,27 @@ def run_avocado_tests(machine, avocado_tests, print_failed=True, env=[]):
 
 def copy_avocado_logs(machine, title):
     if machine and machine.address:
-        dir = "%s-%s.avocado" % (machine.address, title)
         # check if the remote directory exists (if it doesn't, we want to quit silently)
         test_command = "if test -d '{0}'; then echo exists; fi".format(avocado_results_dir)
         if "exists" not in machine.execute(command=test_command,quiet=True):
             return
-        machine.download_dir(avocado_results_dir, dir)
-        if os.path.exists(dir):
-            # avocado creates a "latest" symlink, we can delete that here
-            shutil.rmtree(path=os.path.join(dir, "latest"), ignore_errors=True)
-            print "Avocado results copied to %s" % (dir)
-            for screenshot in glob.glob(os.path.join(dir,'*.png')):
-                shutil.move(screenshot, arg_attachments_dir)
-            # compress the logs and attach that
-            archive = os.path.join(arg_attachments_dir, "{0}.tar.gz".format(dir))
-            subprocess.call(["tar", "cfz", archive, dir])
+
+        # get the screenshots first and move them on the guest so they won't get archived
+        remote_screenshots = os.path.join(avocado_results_dir, "*.png")
+        remote_archive_dir = os.path.split(avocado_results_dir)[0]
+        test_command_screenshots = "if ls '{0}'; then echo exists; fi".format(remote_screenshots)
```

Comment:
How does this work? This is a wildcard glob in a single quoted string. It should never exist unless there's a file actually called ```*.png```. If you've tested this, and it somehow works, then I'll believe you. But worth explaining what's happening here, because it seems esoteric at the very least.